main.go

```diff
@@ -32,7 +33,17 @@ func initialize(unitTestfunc callback, configOverride *config.Config) {
 		fmt.Printf("Failed to create logger: %s\n", err)
 		os.Exit(1)
 	}
-	defer logger.Sync()
+	ticker := time.NewTicker(time.Second)
+	go func() {
+		for {
+			<-ticker.C
+			logger.Sync()
+		}
+	}()
+	defer func() {
+		ticker.Stop()
+		logger.Sync()
+	}()
```

Comment:
With a stop channel and WaitGroup this would just become:
```go
gui.wg.Add(1)
go func() {
    defer gui.wg.Done()
    for {
        select {
        case <-ticker.C:
            logger.Sync()
        case <-stop:
            logger.Sync()
            return
        }
    }
}()
```

Reply:
The initial idea of adding that code was to move the logger initialization and setup code to one place. And, the logger syncing does not need to directly depend on gui. In fact, the logger should function (and be synced) even after `gui.Render()` returned, until the end of the `initialize()` function. So, IMO there is no need to wait for `gui` there.